spyder/plugins/ipythonconsole/plugin.py

```diff
@@ -844,11 +844,9 @@ def interpreter_versions(self):
             import subprocess
             versions = {}
             pyexec = CONF.get('main_interpreter', 'executable')
-            py_cmd = u'%s -c "import sys; print(sys.version)"' % pyexec
-            ipy_cmd = (
-                u'%s -c "import IPython.core.release as r; print(r.version)"'
-                % pyexec
-            )
+            py_cmd = u'env -i %s -c "import sys; print(sys.version)"' % pyexec

```

Comment:
Unless this works on windows, this is a no go

Reply:
Ahh, yes, of course. I don't know if there is an equivalent on windows and a quick google search didn't reveal anything. Since this is only relevant for the Mac app, and should work for *nix systems, perhaps just a control statement here?

Reply:
Sure, anywhere you are using the `env` command there should be a `if os.name != 'nt'` or similar.

I would argue that if some things are really specific to the MacApp and how it works (or does not work) then we should probably consider adding a `IS_MAC_APP` somewhere and only add those changes conditionally as well.

Reply:
Thanks to @ccordoba12 for bringing `running_in_mac_app` to my attention (#12271); I'll keep that in mind in the future. As I mentioned there, I think the benefits extend beyond the Mac app so I think the flow control option is best here.

Reply:
I'm backtracking from my last statement. I'll use `running_in_mac_app` for the cases in this PR. I double-checked and they only apply when launched from the mac app. Launching from non-conda environment (at least in bootstrap) does not cause the issue.